### PR TITLE
home-assistant: 0.114.3 -> 0.114.4, also: samsungtv and ipp integration init/update

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

A few updates and inits to the home-assistant ecosystem.

- New upstream release https://github.com/home-assistant/core/releases/tag/0.114.4
- Add `samsungtv` integration support https://www.home-assistant.io/integrations/samsungtv/
- Update `parse_requirements.py` to also find packages that define `extra_requires`, like `samsungctl[websocket]`, cc #62833
- Update `pyipp` to fix dependency on bumped yarl (#96570)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg test home-assistant

**Comment by jonringer** [comment]
@GrahamcOfBorg test home-assistant

File: pkgs/servers/home-assistant/component-packages.nix [hunk 1]
```diff
@@ -683,7 +683,7 @@
     "safe_mode" = ps: with ps; [ aiohttp-cors hass-nabucasa]; # missing inputs: home-assistant-frontend
     "saj" = ps: with ps; [ ]; # missing inputs: pysaj
     "salt" = ps: with ps; [ ]; # missing inputs: saltbox
-    "samsungtv" = ps: with ps; [ ]; # missing inputs: samsungctl[websocket] samsungtvws[websocket]
+    "samsungtv" = ps: with ps; [ samsungctl samsungtvws];
```
Comment (jonringer) [inline pkgs/servers/home-assistant/component-packages.nix:686]: nit
```suggestion
    "samsungtv" = ps: with ps; [ samsungctl samsungtvws ];
```

File: pkgs/development/python-modules/samsungtvws/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, buildPythonPackage, fetchPypi, isPy27
+, requests
+, websocket_client }:
```
Comment (jonringer) [inline pkgs/development/python-modules/samsungtvws/default.nix:3]: for diffs, nit
```suggestion
, websocket_client
}:
```

File: pkgs/development/python-modules/samsungctl/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildPythonPackage, fetchPypi
+, websocket_client }:
```
Comment (jonringer) [inline pkgs/development/python-modules/samsungctl/default.nix:2]:
```suggestion
, websocket_client
}:
```

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:686]: Yeah, that's something we need to fix at some point in the `parse_requirements.py` script. That diff is autogenerated.

File: pkgs/development/python-modules/samsungctl/default.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/development/python-modules/samsungctl/default.nix]: Fixed.

File: pkgs/development/python-modules/samsungtvws/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/samsungtvws/default.nix]: Fixed.

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix]: Fixed.

**Comment by jonringer** [comment]
that's a big diff, but I think it's still an improvement

**Comment by jonringer** [comment]
@GrahamcOfBorg test home-assistant

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96570 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>8 packages built:</summary><br>- home-assistant<br>- python27Packages.samsungctl<br>- python37Packages.pyipp<br>- python37Packages.samsungctl<br>- python37Packages.samsungtvws<br>- python38Packages.pyipp<br>- python38Packages.samsungctl<br>- python38Packages.samsungtvws</details>

**Pull request merged**